### PR TITLE
WIP: rtnetlink/addr: Support adding arbitrary IP address

### PR DESCRIPTION
**Pull request by cathay4t**
Introducing `AddressHandle::add_raw()` allowing user to add IP
address using netlink `AddressMesssage`.

The most useful options user could be benefit from would be:
 * `valid_lft` and `preferred_lft`. Widely used in DHCP/IPv6-RA.
 * metric

### CODE REVIEW

**Comment by cathay4t** [comment]
@little-dude The test would be slightly complex(need root, need a test veth interface). If you need it be there, I can try.
I intend to use this function in https://github.com/nispor/nispor where we will have test case for this.

**Review by ffmancera — APPROVED** [review]
It looks good to me. I think the best option to test this will be use it from Nispor.. Thanks!

**Comment by cathay4t** [comment]
@little-dude Could you review this PR?

Meanwhile, do we have rule on when a PR can be merged? If not, how about:
 * CI should pass.
 * PR should contain test case unless state otherwise.
 * Approved by 2+ maintainer.

**Comment by little-dude** [comment]
Hey @cathay4t, I agree with your rules :) But maybe for small PRs 1 approval would be enough?

File: rtnetlink/src/addr/add.rs
Comment (little-dude) [inline rtnetlink/src/addr/add.rs:73]: I think it would be nice to document why someone would want to use the `_raw` methods. I'm myself not too sure about it.

Comment (little-dude) [reply rtnetlink/src/addr/add.rs:73]: I mean I understand that you may need to tweak the messages sometimes. But in `LinkRequest` we have [`message_mut()`](https://github.com/little-dude/netlink/blob/3cf41bf7dd658abfdb07e1dd0c52bfaf5035e8f7/rtnetlink/src/link/add.rs#L298) for this. Could we use that here too, for the sake of consistency?

**Comment by little-dude** [comment]
And sorry for taking so long to review this small PR @cathay4t

File: rtnetlink/src/addr/add.rs
Comment (cathay4t) [reply rtnetlink/src/addr/add.rs:73]: Sure. I will use the `message_mut()` way. Thanks.

**Comment by cathay4t** [comment]
Closing as exiting `message_mut()` is enough for this use case.

**Pull request closed without merging**
